include/Grid.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+#pragma once
+
+#include <SDL.h>
+#include "Color.h"
+#include "Rect.h"
+#include "Object.h"
+#include <vector>
+#include <algorithm>
+#include <utility>
+
+namespace ccm
+{
+	class Grid
+	{
+	private:
+		class Tile
+		{
+		public:
+			Tile();
+			Tile(const Rect& pos);
+			Tile(int x, int y, int w, int h);
+
+			Object draw() const;
+
+			void fill();
+			void empty();
+
+		private:
+			Rect m_position;
+			bool m_isFilled;
+		};
+	public:
+		Grid(const Rect& position, int columns, int rows);
+		Grid(int x, int y, int w, int h, int columns, int rows);
+
+		Tile& operator()(int x, int y);
```

Comment:
I think it would be clearer if this was made a `getTile()` function. At the moment it would look like this:
```cpp
Grid grid(...);
Tile& tile = grid(5, 10);
```
As opposed to
```cpp
Grid grid(...);
Tile& tile = grid.getTile(5, 10);
```

Reply:
Agreed.